src/com/jalasoft/search_files/menu/cli_menu_advanced_flow.py

```diff
@@ -10,176 +11,303 @@
         if self.validator.is_number(option) and self.validator.is_the_range(option, self.cli_menu.get_dictionary_size(
                 self.cli_menu.advanced_search_for_size_items) - 1, 1):
+            logger.debug("The value returned is: %s", True)
+            logger.info("Ending the method")
             return True
         else:
+            logger.debug("The value returned is: %s", False)
+            logger.info("Ending the method")
             return False
 
     def advanced_search_for_size_criteria_validation(self, option):
+        logger.info("Starting the method")
         return True
-        #if self.validator.has_valid_characters(option):
-        #    return True
-        #else:
-        #    return False
+        # if self.validator.has_valid_characters(option):
+        #     logger.debug("The value returned is: %s", True)
+        #     logger.info("Ending the method")
+        #     return True
+        # else:
+        #     logger.debug("The value returned is: %s", False)
+        #     logger.info("Ending the method")
+        #     return False
 
     def advanced_search_for_owner_criteria_validation(self, option):
+        logger.info("Starting the method")
         return True
-        #if self.validator.has_valid_characters(option):
-        #    return True
-        #else:
-        #    return False
+        # if self.validator.has_valid_characters(option):
+        #     logger.debug("The value returned is: %s", True)
+        #     logger.info("Ending the method")
+        #     return True
+        # else:
+        #     logger.debug("The value returned is: %s", False)
+        #     logger.info("Ending the method")
+        #     return False
 
     def advanced_search_for_date_menu_validation(self, option):
+        logger.info("Starting the method")
         if self.validator.is_number(option) and self.validator.is_the_range(option, self.cli_menu.get_dictionary_size(
                 self.cli_menu.advanced_search_for_date_items) - 1, 1):
+            logger.debug("The value returned is: %s", True)
+            logger.info("Ending the method")
             return True
         else:
+            logger.debug("The value returned is: %s", False)
+            logger.info("Ending the method")
             return False
 
     def advanced_search_for_date_criteria_validation(self, option):
+        logger.info("Starting the method")
         return True
-        #if self.validator.has_valid_characters(option):
-        #    return True
-        #else:
-        #    return False
+        # if self.validator.has_valid_characters(option):
+        #     logger.debug("The value returned is: %s", True)
+        #     logger.info("Ending the method")
+        #     return True
+        # else:
+        #     logger.debug("The value returned is: %s", False)
+        #     logger.info("Ending the method")
+        #     return False
 
     def advanced_search_for_content_criteria_validation(self, option):
```

Comment:
this method is not yet implemented for this reason the unit test fails
`def test_advanced_search_for_content_criteria_validation_returns_false_when_option_provided_has_non_valid_chars(self):`